### PR TITLE
feat: new methods for loan interest

### PR DESCRIPTION
**Pull request by talamcol**
References [CAR-9003](https://autoricardo.atlassian.net/browse/CAR-9003)

## Motivation and context

[Describe the problem we're solving and lay out the solution]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -256,6 +258,8 @@ Also accompanying modes and param types, as well as default values, are exported
   - [`sendBuyNowApplication`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Buy%20Now%20Application)
   - [`fetchBuyNowConfiguration`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Buy%20Now%20Configuration)
   - [`markBuyNowApplicationAsPaid`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Buy%20Now%20Application/markAsPaidUsingPOST)
+- [Loan](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Loan%20Interest)
+  - [`createLoanInterest`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Loan%20Interest/createUsingPOST)
```
Comment (Averethel) [inline README.md:262]: Don't we need docs for `calculateMonthlyRate` too?

File: src/index.ts [hunk 2]
```diff
@@ -31,10 +31,15 @@ export {
   DealerAnalytics,
   BuyNowConfiguration,
   MessageLeadListing,
+  EmploymentType,
+  Gender,
```
Comment (Averethel) [inline src/index.ts:35]: Do we ask for gender in any of the forms in listings? IIRC it was used in MBG, but that's gone, but online purchase or buyer protection? If yes let's align listings and use exposed enum there.

File: src/types/models/listing.ts [hunk 3]
```diff
@@ -132,6 +132,7 @@ export interface Listing
   interior: string
   kiloWatts: number
   leasingMonthlyRate: number
+  loanMonthlyRate: number
```
Comment (Averethel) [inline src/types/models/listing.ts:135]: Shouldn't this be optional?

File: src/types/models/listing.ts [hunk 4]
```diff
@@ -201,6 +202,7 @@ interface BaseSearchListing {
   modelKey: string
   price: number
   leasingMonthlyRate: number
+  loanMonthlyRate: number
```
Comment (Averethel) [inline src/types/models/listing.ts:205]: Shouldn't this be optional?

**Review by Averethel — APPROVED** [review]
Looks good to me. Please do a sanity check on my questions before merging.

File: src/types/models/listing.ts (hunk 3 above)
Comment (mreljin) [reply src/types/models/listing.ts:135]: BE sends both `leasingMonthlyRate` and `loanMonthlyRate`. if `leasingMonthlyRate` exists -> the user is ineligible for a loan and `loanMonthlyRate` is null. And the other way around. Confirmed with BE

File: src/index.ts (hunk 2 above)
Comment (mreljin) [reply src/index.ts:35]: > but online purchase or buyer protection?

Checked, neither use gender in any form

**Pull request merged**